test/common/testlib.py

```diff
@@ -351,38 +351,84 @@ def mousedown(self, selector):
         self.mouse(selector + ":not([disabled]):not([aria-disabled=true])", "mousedown", 0, 0, 0)
 
     def val(self, selector):
+        """Get the value attribute of a selector
```

Comment:
Please consistently add a period to short description (or consistently drop it).

Also, this description does not really add any extra information. Does that somehow look weird in pydoc if there is no docstring?

Also, can we do the `:type` with proper type annotations instead: `def val(self, selector: str)` is more declarative and useful (mypy!)

Reply:
Typing exists since Python `3.5`, it seems RHEL 8 has `3.6` so that should be possible. It also looks way nicer at least in my editor, which is my primary use case.

![image](https://user-images.githubusercontent.com/67428/176171820-c172b7ff-b543-4bda-90b5-b9a301f5f041.png)

Sadly I realised neovim's LSP cannot determine in a TestClass what self.browser is as it's initially `None` 

![image](https://user-images.githubusercontent.com/67428/176174597-63f82003-2ebb-427c-85e2-23c7b740a469.png)

Not sure if typing will help here.

pydoc out of the box is broken and annoying to fix:
```
cd test/common
pydoc -b
```

```
pydoc.ErrorDuringImport: problem in testlib - ModuleNotFoundError: No module named 'testvm'
```
So either you need some import magic and fiddle every time we change `testvm` which no one will test :)

Reply:
Ok scratch that, I want `typing` now, we just need to teach `MachineCase` that `self.browser` can be a `Browser` instance:

![image](https://user-images.githubusercontent.com/67428/176177941-ead366dc-f8b4-4a16-b2b1-05a24dd866d5.png)

```
@@ -1357,7 +1356,9 @@ class MachineCase(unittest.TestCase):
         if opts.address and self.provision is not None:
             raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")

-        self.browser = None
+        from typing import Type, Optional
+
+        self.browser: Optional[Browser] = None
```

Reply:
This is partial success, as my LSP does not know where `testlib` is, in `test/verify/check-connection` which we ought to really fix:

![image](https://user-images.githubusercontent.com/67428/176178347-c6af1887-31d3-40e7-950d-8d58339c051c.png)

Reply:
So it seems that pyright (my LSP) does not resolve our `sys.path` paths?! annoying!

Reply:
Creating a new file next to `testlib.py` and import `MachineCase` with a simple class that inherits `MachineCase` then `self.browser` is Unknown | None, so do I not understand Python inheritance or why does it work when I inherit `MachineCase` in `testlib.py`.

Reply:
And nevermind, this works now with the `Optional` type in testlib.py and making your editor able to include the correct git worktree. Even in cockpit-podman!

![image](https://user-images.githubusercontent.com/67428/176435338-765ec6c4-41da-4199-9d1a-e1ccd0cd959b.png)


![image](https://user-images.githubusercontent.com/67428/176435393-409a32fc-4ebf-4901-9cf0-6e31d822ee4c.png)